### PR TITLE
Add type annotations for Google mlengine_operator_utils.py

### PR DESCRIPTION
**Pull request by coopergillan**
Add type annotations to `mlengine_operator_utils.py`, including a few changes to ensure the right types are passed through. Specifically, if region is not given, it must be provided in the DAG's `default_args` or else a `KeyError` will now be raised.

I am trying to slowly chip away at the modules with little or no coverage per the command given in #9708. This one was at the top of the list =)

related: #9708

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/cloud/utils/mlengine_operator_utils.py [hunk 1]
```diff
@@ -24,31 +24,35 @@
...
-                        region=None,
-                        dataflow_options=None,
-                        model_uri=None,
-                        model_name=None,
-                        version_name=None,
-                        dag=None,
+T = TypeVar("T", bound=Callable)  # pylint: disable=invalid-name
+
+
+def create_evaluate_ops(task_prefix: str,  # pylint: disable=too-many-arguments
```
Comment (coopergillan) [inline airflow/providers/google/cloud/utils/mlengine_operator_utils.py:42]: Not really sure why the diff showed up like this 🤷‍♂️

File: airflow/providers/google/cloud/utils/mlengine_operator_utils.py [hunk 2]
```diff
@@ -203,7 +210,7 @@ def validate_err_and_count(summary):
     if dag is not None and dag.default_args is not None:
         default_args = dag.default_args
         project_id = project_id or default_args.get('project_id')
-        region = region or default_args.get('region')
+        region = region or default_args['region']
```
Comment (coopergillan) [inline airflow/providers/google/cloud/utils/mlengine_operator_utils.py:213]: When `default_args.get('region')` was being used, `mypy` said that the type was a `Union[str, Any, None]`.

This change also happens to have the fortunate side effect of [now matching what was already in the docstring](https://github.com/apache/airflow/blob/f618cdd19a5cdf09262436978817934d27668997/airflow/providers/google/cloud/utils/mlengine_operator_utils.py#L151).

Comment (ephraimbuddy) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:213]: I would suggest `default_args.get("region", None)`

Comment (coopergillan) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:213]: I think region is actually required in the underlying operator and is expecting a string. Passing None as the second argument here is actually already the default behavior, too.

Comment (turbaszek) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:213]: Right, the `region` is necessary for `MLEngineStartBatchPredictionJobOperator`

Comment (coopergillan) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:213]: @turbaszek - if I'm reading the code correctly, that means that `None` won't work. Hence calling `default_args['region']` directly.

**Review by turbaszek — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**